lib/logthrdestdrv.c

```diff
@@ -886,7 +944,16 @@ log_threaded_dest_driver_init_method(LogPipe *s)
   if (!self->seq_num)
     init_sequence_number(&self->seq_num);
 
-  return _start_worker_thread(self);
+  /* free previous workers array if set to cope with num_workers change */
+  g_free(self->workers);
+  self->workers = g_new0(LogThreadedDestWorker *, self->num_workers);
+
+  gboolean startup_success = TRUE;
+  for (gint i = 0; startup_success && i < self->num_workers; i++)
+    {
+      startup_success &= _start_worker_thread(self);
```

Comment:
The deinit method needs similar code for the those threads that are successfully initialized. self->workers_started might help with that.

Reply:
workers are automatically terminated at exit, so I think they are cleared up just fine.

main_loop_create_worker_thread() gets a function callback that is invoked when a thread needs to exit due to a reload/stop being processed. In that case _request_worker_exit() gets called, which sets a variable and posts an iv_event.

Then the mainloop mechanism will wait for all worker threads to shut down and that's when deinit() gets called. By that time workers exited.

I think that addresses the issue, do you agree?

Reply:
I was thinking that the individual workers also have init deinit mechanism, but that's not the case as only the "master" has init-deinit, hence was my confusion. I agree. Marking as resolved.